mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/ConfigLookUp.java

```diff
@@ -0,0 +1,48 @@
+package org.arquillian.smart.testing.mvn.ext;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.maven.execution.MavenExecutionRequest;
+import org.apache.maven.execution.MavenSession;
+
+import static org.arquillian.smart.testing.configuration.ConfigurationLoader.SMART_TESTING_YAML;
+import static org.arquillian.smart.testing.configuration.ConfigurationLoader.SMART_TESTING_YML;
+
+class ConfigLookUp {
+
+    private final MavenExecutionRequest mavenExecutionRequest;
+    private final File executionDir;
+
+    ConfigLookUp(MavenSession session) {
+        this.mavenExecutionRequest = session.getRequest();
+        this.executionDir = new File(session.getExecutionRootDirectory());
+    }
+
+    File getFirstDirWithConfigOrProjectRootDir() {
+        return getFirstDirWithConfigOrProjectRootDir(executionDir,
+            new File(mavenExecutionRequest.getSystemProperties().getProperty("env.MAVEN_PROJECTBASEDIR")));
```

Comment:
What is the difference between your version and just getting the system enviroment: `System.getenv("MAVEN_PROJECTBASEDIR")` ?

Reply:
btw, what was wrong with the original way of getting the parent directory: `session.getRequest().getMultiModuleProjectDirectory()`?

Reply:
We can use `System.getenv("MAVEN_PROJECTBASEDIR")`. There is no difference.

We don't have `multiModuleProjectDirectory` field in maven version `3.2.5`. So `session.getRequest().getMultiModuleProjectDirectory()` will fail for maven version 3.2.5